### PR TITLE
[SB-25] streamable song link

### PR DESCRIPTION
**Pull request by imariel2d**
Implemented a way to fetch the streamable song link from dropbox, then make it downlable so audio DOM can play it 

also i updated 2 components so it could work, re-ordered audio component variables just to have them ordered :) 

**Evidence** 
https://www.loom.com/share/75cb6258ee8f45b7b363ba073846c7cb

**Jira link**
 https://songbox-io.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=SB&modal=detail&selectedIssue=SB-25

### CODE REVIEW

File: src/lib/apiFetchStreamableSong.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import Promise from 'bluebird';
+import { getWithAuth } from './apiService';
+
+export const apiFetchStreamableSongs = (token, path) => new Promise((resolve, reject) => {
+  getWithAuth(`/api/file${path}`, token, resolve, reject);
```
Comment (tonymtz) [inline src/lib/apiFetchStreamableSong.js]: Abstract this path into a constant :)

File: src/store/actions/playerActions.js [hunk 2]
```diff
@@ -5,7 +5,22 @@ import {
...
 } from '../constants';
 
+import { apiFetchStreamableSongs } from '../../lib/apiFetchStreamableSong';
+
 export const toggleRepeat = (payload) => ({ type: PLAYER_SET_REPEAT, payload });
 export const setVolume = (payload) => ({ type: PLAYER_SET_VOLUME, payload });
 export const setSong = (payload) => ({ type: PLAYER_SET_CURRENT_SONG, payload });
 export const setSongLink = (payload) => ({ type: PLAYER_SET_SONG_LINK, payload });
+
+export const fetchStreamableSong = (path) => (dispatch, getState) => {
```
Comment (tonymtz) [inline src/store/actions/playerActions.js]: nit: `getSongStreamLink`

File: src/lib/apiFetchStreamableSong.js (hunk 1 above)
Comment (arielplas) [reply src/lib/apiFetchStreamableSong.js]: do you mean like a variable?
**const routeWithPath = /api/file${path};**

or a constant in routes.js?
**export const API_FILE_ROUTE = `/api/file`**

Comment (tonymtz) [reply src/lib/apiFetchStreamableSong.js]: a constant in routes.js :)

File: src/routes.js [hunk 3]
```diff
@@ -9,3 +9,6 @@ export const FILE_PATH = '/app/:path+';
 export const FAVORITES_PATH = '/favorites';
 export const SETTINGS_PATH = '/settings';
 export const HELP_PATH = '/help';
+
+/* API routes */
+export const API_FILE_ROUTE = '/api/file';
```
Comment (tonymtz) [inline src/routes.js:14]: 👏🏼

**Review by tonymtz — APPROVED** [review]
👍🏼

**Pull request merged**